### PR TITLE
set difference

### PR DESCRIPTION
**Pull request by anurag629**
#14226

- [ ] Get the function base case working in backends
    - [x] Numpy
    - [x] Jax
    - [x] Torch
    - [ ] Paddle
    - [x] Tensorflow
- [ ] Add superset behaviour for all framework cases

### CODE REVIEW

File: ivy/functional/ivy/experimental/set.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+)
+from ivy.utils.exceptions import handle_exceptions
+
+
+@to_native_arrays_and_back
+@handle_out_argument
+@handle_array_like_without_promotion
+@handle_nestable
+@handle_exceptions
+def difference(
```
Comment (hirwa-nshuti) [inline ivy/functional/ivy/experimental/set.py:21]: I think we can make a superset for this function as it is handled differently on the Tensorflow side, we can add `assume_unique` argument. And handle the case of multi-dimensional arrays as `TensorFlow supports this.
```python
import tensorflow as tf

set1 = tf.sparse.SparseTensor(indices=[[0, 0], [1, 2], [2, 3]], values=[1, 2, 3], dense_shape=[3, 4])
set2 = tf.sparse.SparseTensor(indices=[[0, 0], [2, 2], [2, 3]], values=[1, 2, 3], dense_shape=[3, 4])

diff = tf.sets.difference(set1, set2)
print(diff)
```

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @anurag629 

I added some few comments below check them out🙂
Adding that we will need paddle implementation as well😁

File: ivy/data_classes/container/experimental/set.py [hunk 2]
```diff
@@ -1,5 +1,126 @@
...
+
+# local
 from ivy.data_classes.container.base import ContainerBase
+import ivy
 
 
 class _ContainerWithSetExperimental(ContainerBase):
-    pass
+    @staticmethod
+    def static_difference(
```
[the remark this answers is not in the document]
Comment (anurag629) [reply ivy/data_classes/container/experimental/set.py:11]: But other functions are not private(in other files).

[the remark this answers is not in the document]
Comment (hirwa-nshuti) [reply ivy/data_classes/container/experimental/set.py:11]: Only the container static methods

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**